File: java/Longest_Common_Prefix.java
// compare strs[1~n-1] with strs[0]
// time is O(mn), where m is the average length of strs and n is the String[] length. 
public class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length==0) return "";
        if (strs.length==1) return strs[0] ;
        String target = strs[0];
        int i = 0;
        for (; i < target.length() ; ++i) {
            for (int j = 1 ; j < strs.length ; ++j) {
                if (strs[j].length() <= i || strs[j].charAt(i) != target.charAt(i))
                    return target.substring(0,i);
            }
        }
        return target;
    }
}

File: Python/Longest Common Prefix.py
class Solution:
    # @param {string[]} strs
    # @return {string}
    def longestCommonPrefix(self, strs):
        if len(strs) == 0:
            return ""
        elif len(strs) == 1:
            return strs[0]
        s1 = strs[0]
        for s2 in strs:
            s1 = len(s1) <= len(s2) and s1[:] or s1[:len(s2)] #三元操作符的使用
            for i in range(len(s1)):
                if s1[i] != s2[i]:
                    s1 = s1[:i]
                    break
        return s1
